subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/incremental/ClassSetAnalysisUpdater.java

```diff
@@ -65,8 +72,51 @@ public void updateAnalysis(JavaCompileSpec spec) {
         for (File baseDir : baseDirs) {
             fileOperations.fileTree(baseDir).visit(analyzer);
         }
+        if (spec.getCompileOptions().isIncrementalAnnotationProcessing()) {
+            recordGeneratedTypeDependencies(analyzer, spec);
+        }
         ClassSetAnalysisData data = analyzer.getAnalysis();
         stash.put(data);
         LOG.info("Class dependency analysis for incremental compilation took {}.", clock.getElapsed());
     }
+
+    private void recordGeneratedTypeDependencies(ClassFilesAnalyzer analyzer, JavaCompileSpec spec) {
```

Comment:
❌ This could also use some unit testing, just to make sure that each kind of entry is handled correctly.

Reply:
I still need to add this.